virttest/utils_misc.py

```diff
@@ -2795,51 +2843,79 @@ def format_windows_disk(session, did, mountpoint=None, size=None, fstype="ntfs")
+                          disk kname, serial and wwn.
+    :param partition: if true, can format all disks; otherwise,
+                      only format the ones with no partition originally.
     :param mountpoint: mount point for the disk.
+    :param size: partition size.
     :param fstype: filesystem type for the disk.
-    :param ostype: guest os type 'windows' or 'linux'.
     :return Boolean: disk usable or not.
     """
-    list_disk_cmd = "lsblk -p -oKNAME,MOUNTPOINT,SERIAL,SIZE,MODEL,WWN"
-    disks = session.cmd_output(list_disk_cmd)
+    disks = list_linux_guest_disks(session, partition)
     logging.debug("Disks detail: %s" % disks)
-    for disk in disks.splitlines():
-        if did in disk:
-            kname = disk.split()[0]
-            logging.info("Create partition on disk '%s'" % kname)
-            mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
-            session.cmd(mkpart_cmd)
-            logging.info("Format partition to '%s'" % fstype)
-            format_cmd = "yes|mkfs -t %s %s" % (kname, fstype)
-            session.cmd(format_cmd)
-            logging.info("Mount the disk to '%s'" % mountpoint)
-            mount_cmd = "mount -t %s %s %s" % (fstype, kname, mountpoint)
-            session.cmd(mount_cmd)
-            return True
+    for line in disks:
+        kname = line.split('/')[2]
+        did_list = all_disks_did[kname]
+        if did not in did_list:
+            # Continue to search target disk
+            continue
+        if not size:
+            size_output = session.cmd_output_safe("lsblk -oKNAME,SIZE|grep %s"
+                                                  % kname)
+            size = size_output.splitlines()[0].split()[1]
+        all_disks_before = list_linux_guest_disks(session, True)
+        devname = line
+        logging.info("Create partition on disk '%s'" % devname)
+        mkpart_cmd = "parted -s %s mklabel gpt mkpart "
+        mkpart_cmd += "primary 0 %s"
+        mkpart_cmd = mkpart_cmd % (devname, size)
+        session.cmd_output_safe(mkpart_cmd)
+        session.cmd_output_safe("partprobe %s" % devname)
+        all_disks_after = list_linux_guest_disks(session, True)
+        partname = (all_disks_after - all_disks_before).pop()
+        logging.info("Format partition to '%s'" % fstype)
+        format_cmd = "yes|mkfs -t %s %s" % (fstype, partname)
+        session.cmd_output_safe(format_cmd)
+        if not mountpoint:
+            session.cmd_output_safe("mkdir /mnt/%s" % kname)
+            mountpoint = os.path.join("/mnt", kname)
+        logging.info("Mount the disk to '%s'" % mountpoint)
+        mount_cmd = "mount -t %s %s %s" % (fstype, partname, mountpoint)
+        session.cmd_output_safe(mount_cmd)
+        return True
 
     return False
 
 
-def format_guest_disk(session, did, mountpoint, size, fstype, ostype):
+def format_guest_disk(session, did, all_disks_did, partition,
+                      mountpoint, size, fstype, ostype):
```

Comment:
How about set fstype=None as default, this way can use it without this value, and format with default fstype of format_linux_disk() and format_windows_disk()